File: sprint5/By lessons/lesson12code3/lesson12code3.swift
func split(root: TreeNode?, k: Int) -> (TreeNode?, TreeNode?) {
    if root == nil {
        // Подумайте, что надо вернуть в таком случае.
        return (nil, nil)
    }
    if get_size(root: root!.left) + 1 <= k {
        var k = k - 1 - get_size(root: root!.left)
        let (right, rightSide) = split(root: root!.right, k: k)
        root!.right = right
        return (root, rightSide)
    }
    // Что должно происходить при спуске рекурсии в левое поддерево?
    let (left, leftSide) = split(root: root!.left, k: k)
    root!.left = left
    return (leftSide, root)
}

File: sprint5/By lessons/lesson12code3/lesson12code3.cs
class Solution {
    public static (TreeNode, TreeNode) Split(TreeNode root, int k) {
        if (root == null) {
            // Подумайте, что надо вернуть в таком случае.
            return (null, null);
        }
        if (Get_Size(root.left) + 1 <= k) {
            k -= 1 + Get_Size(root.left);
            var (right, rightSide) = Split(root.right, k);
            root.right = right;
            return (root, rightSide);
        }
        // Что должно происходить при спуске рекурсии в левое поддерево?
        var (left, leftSide) = Split(root.left, k);
        root.left = left;
        return (leftSide, root);
    }
}
